app/frontend/data-dive/dives/dive-editor.ts

```diff
@@ -86,11 +86,19 @@ export default class DiveEditor extends ContentPart<DiveEditorState> {
         this.listenMessage(QueryForm.settingsChangedKey, m => {
             const query = m.data
             log.info(`Query settings changed`, query)
-            this.queryTabs.updateTab({key: query.id, title: query.name})
+            this.queryTabs.updateTab({ key: query.id, title: query.name })
+        })
+
+        // Reorder queries in the list when the tab sort order is updated.
+        this.listenMessage(TabContainerPart.tabReorderedKey, m => {
+            const { permutation } = m.data
+            try {
+                this.queries = TabContainerPart.reorderByPermutation(this.queries, permutation)
+            } catch (Error) { /* Skip if permutation is out of sync */ }
         })
```

Comment:
I think it'd be better to change `queries` to a `Map`. There are a couple of places that do `this.queries.find(q => q.id == id)` which is a sure sign that a `Map` is the correct data structure anyway. Then we can store the order as an array of ids in the correct order in a new variable `this.queryOrder`. Then this listener can just become

```ts
this.listenMessage(TabContainerPart.tabReorderedKey, m => {
    this.queryOrder = m.data.newTabOrder
})
```

Then we can change places that make use of queries to look up from the Map, and change `serialize` to pull queries from the Map in the order specified by `this.queryOrder`.